doc/vxlan/data_plane_VXLAN_HLD.md

```diff
@@ -0,0 +1,507 @@
+
+
+
+---
+##### Figure 1. Current architecture
+![Current VXLAN ARchitecture](images/curent_vxlan_arch.png)
+---
+
+##### Figure 1. Dataplane architecture
+![Dataplane VXLAN ARchitecture](images\dt_vxlan_arch.png)
+---
+
+#### SAI API
+No changes to SAI APIs are required.
+
+### 2. Configuration and management
+
+#### CLI Enhancements
+
+##### 5.1 Remote VTEP IP configuration
+
+* Ability to add and remove remote VTEP
+   - config vxlan remote add <vtepname> <ipv4-address>
+   - config vxlan remote del <vtepname>
+   - vtepname is a string. 
+   - ipv4-address is an IPV4 address in dotted notation A.B.C.D
+
+##### 5.2 Remote VTEP-VNI association configuration 
+
+* Ability to add and remove remote VTEP assiciation to VNI
+   - config vxlan remote map add <vtepname> <vni-id>
+   - config vxlan remote map del <vtepname> <vni-id>
+   - vtepname is a string. 
+   - ipv4-address is an IPV4 address in dotted notation A.B.C.D
+
+#### Yang Enhancements
+
+##### VXLAN_REMOTE_VTEP
+
+container VXLAN_REMOTE_VTEP {
+
+    description "config db VXLAN_REMOTE_VTEP table";
+
+    list VXLAN_REMOTE_VTEP_LIST {
+
+        key "name";
+        max-elements 1;
+
+        leaf name {
+        /* vni devices are created of the form 'name'-vlanid
+            The kernel has a max limit of 15 chars for netdevices.
+            keeping aside 5 chars for hyphen and vlanid the
+            name should have a max of 10 chars */
+
+            type string {
+                length 1..10;
+            }
+        }
+
+        leaf src_ip {
+            type inet:ip-address;
+        }
+    }
+}
+
+##### VXLAN_REMOTE_VTEP_VNI
+
+container VXLAN_REMOTE_VTEP_VNI {
+
+    description "config db VXLAN_REMOTE_VTEP_VNI table";
+
+    list VXLAN_REMOTE_VTEP_VNI_LIST {
+        key "name mapname";
+
+        leaf name {
+            type leafref {
+                path "/svxlan:sonic-vxlan/svxlan:VXLAN_REMOTE_VTEP_LIST/svxlan:VXLAN_REMOTE_VTEP_VNI_LIST/svxlan:name";
+            }
+        }
+
+        leaf mapname {
+            type string;
+        }
+
+        leaf vni {
+            mandatory true;
+            type stypes:vnid_type;
+        }
+    }
+}
+
+
+#### Config DB Enhancements
+
+##### 2.1 VXLAN_REMOTE_VTEP_TABLE
+
+Producer: config manager
+
+Consumer: VxlanMgr
+
+Description: This table holds the tunnel name specifying the VTEP remote IP used for data plane configured tunnels.
+
+; New table
+
+key = VXLAN_REMOTE_VTEP_TABLE|remote_vtep_name
+                          ; VTEP name as a string
+; field = value
+remote_ip = ipv4_address ; Source Ipv4 address of the remote VTEP.
+
+
+##### 2.2 VXLAN_REMOTE_VTEP_VNI_TABLE
+
+Producer: config manager
+
+Consumer: VxlanMgr
+
+Description: This table specifies mapping of remote VTEP to VNI.
+
+Schema:
+
+; New table
+; specifies mapping of remote VTEP to VNI
+
+key = VXLAN_REMOTE_VTEP_VNI_TABLE|remote_vtep_name|tunnel_map_name
```

Comment:
Shouldnt the key be
VXLAN_REMOTE_VTEP_VNI_TABLE|vtep_name|Vlanname  

The tunnel map name can technically be anything and not reflective of the VLAN and VNI

Reply:
Does the key has to be reflective of vlan/vni ? My understanding is that key only has to be unique and since vni (and maybe vlan) is field in the entry it doesn't have to be specified again in the key.